crates/net/network/src/eth_requests.rs

```diff
@@ -128,13 +127,18 @@ where
                 }
 
                 headers.push(header);
+
                 if headers.len() >= MAX_HEADERS_SERVE {
                     break
                 }
-
-                total_bytes += APPROX_HEADER_SIZE;
-                if total_bytes > SOFT_RESPONSE_LIMIT {
-                    break
+                match headers.encode_max(APPROX_HEADER_SIZE, SOFT_RESPONSE_LIMIT) {
+                    Ok(_) => {
+                        // If encode_max succeeds, continue accumulating headers
+                    }
+                    Err(_) => {
+                        // If encode_max fails,stop the loop
+                        break;
+                    }
```

Comment:
this entirely defeats the purpose of using approximations because now we're spending more time on redundant encoding.

please undo these changes here and to blocks